### DISCUSSION TITLE
The robot cannot complete full movement for short distances but can complete long distances. Why?

### DISCUSSION THREAD

**Discussion by Hazeline2018** [LemLib/LemLib#181] [Q&A]
I'm fine tuning the auton routines. But it seems that robot always fails following the exact movement codes, especially for short distance movements. For example, I have the following sequence of movements that I'm trying to achieve.

`   chassis.moveToPose(-33, -40, -90, 2000, {.forwards = false, .lead = 0.6, .maxSpeed = 80});  
    chassis.moveToPose(-40, -40, -90, 1000, {.maxSpeed = 100});  
    chassis.moveToPose(-33, -40, 90, 1500, {.maxSpeed = 60}, false);  `

Simply put, I want the robot to move to (-33, -40), then backup 7 inches, then turn 180 degrees and move back to the same position (-33, -40). But the actual behavior was that the robot barely backed up, and it turned 180 but did not move back to the starting position. No matter how I adjust the speed and timeout settings, it had little affect to that strange behavior. It felt like as if the robot knows it's going to move back to (-33,-40) at the get go, so it doesn't want to back up and come back. So it just barely moves and only turns 180 degrees.

Not sure if I have explained my problem clearly. But I don't have problem with much longer distance movements.
Can you please help root cause and advise me how to improve?
Thanks,
Terry

**Comment by liam-teale** [comment]
`Chassis::moveToPose()` is not great for short motions or motions with a large change in angular position. Consider using `Chassis::moveToPoint()` and `Chassis::turnToPoint()` for this sequence.

**Reply by Hazeline2018** [reply]
Okay, thanks for the advice and will give it a try!

BTW, would it be a good combination of 'Chassis::moveToPoint()' and 'chassis.turnToHeading()' for this sequence? I felt that 'chassis.turnToHeading()' has higher accuracy than 'Chassis::turnToPoint()' for heading. Obviously, I'd need to calculate the heading angle before hand when using 'chassis.turnToHeading()'.

**Reply by liam-teale** [reply]
whether you use `Chassis::turnToHeading()` or `Chassis::turnToPoint()` depends on what you are trying to accomplish.

For example, you want the robot to face object `X`. Usually, the robot needs to turn about 80 degrees to face object `X`. But what if the robot is not where we expect it to be? Well, when it turns to face 80 degrees it may no longer face `X`. This is where `Chassis::turnToPoint()` comes in. Even if the robot does not start where we expect it to, it will still face `X`.

Sometimes this is not necessary and the heading the robot ends up is more important than an object it faces. For example, when you want to line the robot up to a wall. In this case we'd want to use `Chassis::turnToHeading()` instead.
